alot/account.py

```diff
@@ -254,11 +255,27 @@ def __init__(self, address=None, aliases=None, alias_regexp=None,
             encrypt_by_default = u"none"
             logging.info(msg)
         self.encrypt_by_default = encrypt_by_default
+        # cache alias_regexp regexes
+        if self.alias_regexp != "":
+            self._alias_regexp = re.compile(
+                u'^' + unicode(self.alias_regexp) + u'$',
+                flags=0 if case_sensitive_username else re.IGNORECASE)
 
-    def get_addresses(self):
-        """return all email addresses connected to this account, in order of
-        their importance"""
-        return [self.address] + self.aliases
+    def matches_address(self, address):
+        """returns whether this account knows about an email address
+
+        :param str address: address to look up
+        :rtype: bool
+        """
+        if self.address == address:
```

Comment:
I wonder if we can replace all this stuff about an accounts main address / aliases and stuff it into the `alias_regex`:
A user can then only define that regex, and if she wants multiple addresses that would require to use logical OR's in that re. The test here then reduces to `re.matches`.

The problem with this is just that there may be no finite list of "main addresses" to tab-complete from..

Reply:
There's ui.senderhistory which is already used for some completion, but that's of course empty if the client is freshly set up. I'd keep it the way it is, especially considering not every user is fluent in regex.

Reply:
actually that `senderhistory` is cached in a file and re-read upon ui startup.

Reply:
@lucc @dcbaker opinions on this? I believe that we could get rid of `account.aliases` config option in favor of `account.alias_regexp`. This would simplify the settingsmanager code at the expense of more complicated (super)user config files..

Reply:
What I meant to say was that a first-time user will have no history to complete from (and first-time UX is somewhat important).

Reply:
if the question is just about these few lines of code I would leave them.